Set MySQL max packet to 256mb

AP-Hunt opened the pull request:
What
---
`max_allowed_packet` defaults to 4mb, but Amazon's recommended value [1] is 64mb. This causes issues for MySQL users who are using BLOB type columns, or otherwise have large field values.

We've raised this to 256mb, after a [support request](https://govuk.zendesk.com/agent/tickets/4476087)

The value has been set by hand for existing parameter groups
    
[1] https://aws.amazon.com/blogs/database/best-practices-for-configuring-parameters-for-amazon-rds-for-mysql-part-3-parameters-related-to-security-operational-manageability-and-connectivity-timeout/

How to review
---
1. Code review

AP-Hunt committed:
Refactor AWS RDS parameter group tests to group by database engine

AP-Hunt committed:
Set max_allowed_packet on all new MySQL parameter groups

max_allowed_packet defaults to 4mb, and Amazon's recommended value [1] is 64mb.
We're allowing a value of 256mb in order to enable CMS' like Drupal.

[1] https://aws.amazon.com/blogs/database/best-practices-for-configuring-parameters-for-amazon-rds-for-mysql-part-3-parameters-related-to-security-operational-manageability-and-connectivity-timeout/

whi-tw reviewed (approved):
lgtm - works as expected

whi-tw reviewed (changes requested):
1024 * 64 = kb, not bytes

AP-Hunt committed:
Remove integration tests for Postgres 9.5

AWS RDS no longer offers Postgres 9.5, and as a result it's causing integration
tests to fail.

This commit deliberately DOES NOT remove any code paths concering Postgres 9.5,
because we still have some databases on that engine version.

whi-tw reviewed (approved):
Much better!
